### PR TITLE
Adopt UT test 10051

### PR DESCRIPTION
**Pull request by guy-maurel**
Thanks to Kalyana Chakravarthi

### CODE REVIEW

File: src/space.cpp [hunk 1]
```diff
@@ -1223,6 +1223,17 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
          log_rule("sp_fparen_dbrace");
          return(cpd.settings[UO_sp_fparen_dbrace].a);
       }
+      // To fix issue #1234
+      // check for initializers and add space or ignore based on the option.
+      if (first->parent_type == CT_FUNC_CALL)
+      {
+         chunk_t *tmp = chunk_get_prev_type(first, first->parent_type, first->level);
+         tmp = chunk_get_prev_ncnl(tmp);
+         if (tmp != NULL && tmp->type == CT_NEW)
```
Comment (CDanU) [inline src/space.cpp]: use chunk_is_token() for the CT_NEW check

File: tests/config/UNI-1338.cfg [hunk 2]
```diff
@@ -0,0 +1,8 @@
+sp_fparen_brace                 = add
+indent_columns                  = 4
+indent_with_tabs                = 0
+indent_class                    = true
+nl_class_leave_one_liners       = true
+nl_end_of_file                  = force
+nl_end_of_file_min              = 1
+sp_inside_braces                = add
```
Comment (CDanU) [inline tests/config/UNI-1338.cfg]: This test (config, input, output) does not seem to belong to this PR. The new option is missing here.

File: tests/c-sharp.test [hunk 3]
```diff
@@ -83,8 +83,9 @@
 10036  U14-Cpp.cfg                          cs/inttypes.h.mm
 10039  615_nested_usings.cfg                cs/615_nested-usings.cs
 10044  U04-Cs.cfg                           cs/ifcomment.cs
-10045  U05-Cs.cfg                           cs/UNI-1288.cs
+10045  UNI-1288.cfg                         cs/UNI-1288.cs
 10046  U15-Cpp.cfg                          cs/UNI-1333.mm
+10051  UNI-1338.cfg                         cs/UNI-1338.cs
```
Comment (CDanU) [inline tests/c-sharp.test]: ↑

File: src/space.cpp [hunk 4]
```diff
@@ -1223,6 +1223,17 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
...
+         {
+            return(cpd.settings[UO_sp_fparen_brace_initializer].a);
```
Comment (CDanU) [inline src/space.cpp]: `log_rule("sp_fparen_brace_initializer") ;` missing

File: tests/config/UNI-1288.cfg [hunk 5]
```diff
@@ -0,0 +1,9 @@
+sp_assign                       = add
+sp_fparen_brace_initializer     = add
```
Comment (CDanU) [inline tests/config/UNI-1288.cfg]: This config does have too many options in order to only check `sp_fparen_brace_initializer`

Comment (guy-maurel) [reply tests/config/UNI-1288.cfg]: It use the default value of the new option.

Comment (CDanU) [reply tests/config/UNI-1288.cfg]: I am not entirely sure what you mean by that. 
I wanted to point out that most of the  options are not sql to test the behavior of the new option, esp. options like: 

```sql
indent_columns                  = 4
indent_with_tabs                = 0
indent_class                    = true
```

are a big annoyance that is spreading throughout almost all tests. We should maybe consider to set those to default values in Uncrustify itself.  
I have never seen anyone that uses the current 8 space default indent (and spaces are '__*obviously*__' better than tabs).

Comment (mihaipopescu) [reply tests/config/UNI-1288.cfg]: As much as I would like those to be the new defaults changing them it's going to break backwards compat.

Comment (gmaurel) [reply tests/config/UNI-1288.cfg]: I think we may not change the defaults, even if they have a "wrong" value.

Comment (CDanU) [reply tests/config/UNI-1288.cfg]: OK, we will discuss the default values issue somewhere else. 
You can however change the tests to that the unneeded options aren't used anymore.
